src/app/forms/custom-sections/psv-brakes/psv-brakes.component.ts

```diff
@@ -0,0 +1,107 @@
+import { CustomFormGroup, FormNode, FormNodeEditTypes, FormNodeWidth } from '@forms/services/dynamic-form.types';
+import { MultiOptionsService } from '@forms/services/multi-options.service';
+import { PsvBrakesTemplate } from '@forms/templates/psv/psv-brakes.template';
+import { Brake, ReferenceDataResourceType } from '@models/reference-data.model';
+import { Brakes, TechRecordModel } from '@models/vehicle-tech-record.model';
+import { Store } from '@ngrx/store';
+import { ReferenceDataState, selectBrakeByCode } from '@store/reference-data';
+import { debounceTime, mergeMap, Observable, of, Subject, takeUntil, withLatestFrom } from 'rxjs';
+
+@Component({
+  selector: 'app-psv-brakes',
+  templateUrl: './psv-brakes.component.html',
+  styleUrls: ['./psv-brakes.component.scss']
+})
+export class PsvBrakesComponent implements OnInit, OnChanges, OnDestroy {
+  @Input() vehicleTechRecord!: TechRecordModel;
+  @Input() isEditing = false;
+
+  @Output() formChange = new EventEmitter();
+
+  form!: CustomFormGroup;
+  template!: FormNode;
+
+  selectedBrake?: Brake;
+
+  private destroy$ = new Subject<void>();
+
+  constructor(
+    private dfs: DynamicFormService,
+    private optionsService: MultiOptionsService,
+    private referenceDataStore: Store<ReferenceDataState>
+  ) {}
+
+  ngOnInit(): void {
+    this.form = this.dfs.createForm(PsvBrakesTemplate, this.vehicleTechRecord) as CustomFormGroup;
+
+    this.form.cleanValueChanges
+      .pipe(
+        debounceTime(400),
+        takeUntil(this.destroy$),
+        mergeMap((event: any) => event?.brakes?.brakeCode ? this.referenceDataStore.select(selectBrakeByCode(event.brakes.brakeCode)) : of()),
+        withLatestFrom(this.form.cleanValueChanges)
+      )
+      .subscribe(([selectedBrake, event]: ([Brake | undefined, any])) => {
+        // Set the brake details automatically based selection
+        const brakes = event?.brakes as Brakes;
+
+        if (selectedBrake && brakes?.brakeCode) {
+          event.brakes['dataTrBrakeOne'] = selectedBrake.service;
+          event.brakes['dataTrBrakeTwo'] = selectedBrake.secondary;
+          event.brakes['dataTrBrakeThree'] = selectedBrake.parking;
+        }
+
+        this.formChange.emit(event)
+      });
+
+      this.optionsService.loadOptions(ReferenceDataResourceType.Brake);
+  }
+
+  ngOnChanges(changes: SimpleChanges): void {
+    const { vehicleTechRecord } = changes;
+
+    if (this.form && vehicleTechRecord?.currentValue && vehicleTechRecord.currentValue !== vehicleTechRecord.previousValue) {
+      this.form.patchValue(vehicleTechRecord.currentValue, { emitEvent: false });
+    }
+  }
+
+  ngOnDestroy(): void {
+    this.destroy$.next();
+    this.destroy$.complete();
+  }
+
+  get brakesForm(): FormGroup {
+    return this.form.get('brakes') as FormGroup;
+  }
+
+  get booleanOptions(): MultiOptions {
+    return [{ value: true, label: 'Yes' }, { value: false, label: 'No' }];
+  }
+
+  get retarderOptions(): MultiOptions {
```

Comment:
Do we not have an enum for retarder options? is it not something we can pass through that enum to options util func?

Reply:
I found the `Retarder` `enum` and I am using that instead now.

Thanks for the heads up.